src/wp-includes/pluggable.php

```diff
@@ -2603,90 +2603,129 @@ function wp_hash( $data, $scheme = 'auth' ) {
 	 * instead use the other package password hashing algorithm.
 	 *
 	 * @since 2.5.0
+	 * @since x.y.z The password is now hashed using bcrypt instead of phpass.
 	 *
-	 * @global PasswordHash $wp_hasher PHPass object.
+	 * @global PasswordHash $wp_hasher phpass object.
 	 *
 	 * @param string $password Plain text user password to hash.
 	 * @return string The hash string of the password.
 	 */
 	function wp_hash_password( $password ) {
 		global $wp_hasher;
 
-		if ( empty( $wp_hasher ) ) {
-			require_once ABSPATH . WPINC . '/class-phpass.php';
-			// By default, use the portable hash from phpass.
-			$wp_hasher = new PasswordHash( 8, true );
+		if ( ! empty( $wp_hasher ) ) {
+			return $wp_hasher->HashPassword( trim( $password ) );
 		}
 
-		return $wp_hasher->HashPassword( trim( $password ) );
+		/**
+		 * Filters the options passed to the password_hash() and password_needs_rehash() functions.
+		 *
+		 * @since x.y.z
+		 *
+		 * @param array $options Array of options to pass to the password hashing functions.
+		 *                       By default this is an empty array which means the default
+		 *                       options will be used.
+		 */
+		$options = apply_filters( 'wp_hash_password_options', array() );
+
+		return password_hash( trim( $password ), PASSWORD_BCRYPT, $options );
```

Comment:
It would be advisable to prevent the 72-byte truncation by design.

Consider this:

```php
$prehashed = base64_encode( hash_hmac,( 'sha512', trim( $password ), CONFIG_SALT_GOES_HERE, true ) );
return password_hash( $prehashed, PASSWORD_BCRYPT, $options );
```

You can verify that this is safer than a naked `password_hash()` with a simple unit test:

* Hash the letter `A` repeated 72 times.
* Attempt to verify a password consisting of the letter `A` repeated 73 times.

## Why hash_hmac()?

This binds the prehash to a server-side secret, that can exist alongside the [usual WordPress secrets](https://api.wordpress.org/secret-key/1.1/salt/). In essence, without this secret, passwords cannot even begin to be cracked.

## Why SHA-512 + Base64?

SHA-512 produces 64 bytes of output, which is just shy of the 72 byte limit for bcrypt. However, there can be a `NUL` character in the midst, so it could truncate early if not encoded.

By base64-encoding the SHA-512 hash (really, HMAC output), you prevent `NUL` characters. This is the optimal trade-off.

* By itself, SHA-512 hashes have a collision resistance of 256 bits.
* The base64-encoded hash will be about 86 bytes (88 with padding).
* Although bcrypt silently truncates the last 14 (16 with padding) characters, we still have 72 base64-encoded characters (about 432 bits of security).
* Each WordPress install will have a different HMAC key, which will frustrate crackers tremendously (especially if they can only leak the contents of the SQL database, not the filesystem.

I believe this small bit of complexity is worth the obvious security benefits.

## Can we do it later?

I don't believe we can. We either adopt this strategy as part of the phpass migration, or we don't do it ever.